lib/affirm/charge_event.rb

```diff
@@ -0,0 +1,20 @@
+module Affirm
+  class ChargeEvent
+    attr_reader :id, :transaction_id, :type, :created,
+      :amount, :fee, :order_id, :shipping_carrier, :shipping_confirmation,
+      :fee_refunded
+
+    def initialize(attrs)
```

Comment:
No Hashr or method_missing will be allowed here. 

Can't grep for method names and not easy to ready. The idea of having a hash act as a hash/object/model thingy has caused us a load of issues in other places.

I prefer this way of doing it.